detection_rules/utils.py

```diff
@@ -326,14 +328,16 @@ def load_rule_contents(rule_file: Path, single_only=False) -> list:
         return contents or [{}]
     elif extension == '.toml':
         rule = pytoml.loads(raw_text)
-    else:
-        rule = load_dump(rule_file)
+    elif extension.lower() in ('yaml', 'yml'):
+        rule = load_dump(str(rule_file))
```

Comment:
Question: Loading yaml is handled in `load_dump`, why do we include it here? Maybe we want to refer to load_dump for deciphering format and loading since it is called throughout the code?

Reply:
I think there is some historical context why we do not use `load_dump` for Kibana exports (typically `.json`) from https://github.com/elastic/detection-rules/pull/83 and and issue for using `load_dump` for toml files from https://github.com/elastic/detection-rules/pull/1073

I think it makes sense to have a single function to handle loading like you suggest, but I am not sure if that is possible here, unless we want to not use `load_dump` from eql at all.

Reply:
@terrancedejesus essentially it was an implicit way to bypass any other extension types (like MD)

Reply:
Lets drop the `allow_empty_rule` and just return an empty list

```suggestion
    elif extension.lower() in ('yaml', 'yml'):
        rule = load_dump(str(rule_file))
    else:
        return []
```